ClientCore/CnCNet5/GameCollection.cs

```diff
@@ -15,128 +16,150 @@ public class GameCollection
     {
         public List<CnCNetGame> GameList { get; private set; }
 
-        public void Initialize(GraphicsDevice gd)
+        public void Initialize()
         {
             GameList = new List<CnCNetGame>();
 
+            var assembly = Assembly.GetAssembly(typeof(GameCollection));
+            using Stream dtaiconStream = assembly.GetManifestResourceStream("ClientCore.Resources.dtaicon.png");
+            using Stream tiiconStream = assembly.GetManifestResourceStream("ClientCore.Resources.tiicon.png");
+            using Stream tsiconStream = assembly.GetManifestResourceStream("ClientCore.Resources.tsicon.png");
+            using Stream moiconStream = assembly.GetManifestResourceStream("ClientCore.Resources.moicon.png");
+            using Stream yriconStream = assembly.GetManifestResourceStream("ClientCore.Resources.yricon.png");
+            using Stream rriconStream = assembly.GetManifestResourceStream("ClientCore.Resources.rricon.png");
+            using Stream cncriconStream = assembly.GetManifestResourceStream("ClientCore.Resources.cncricon.png");
+            using Stream cncneticonStream = assembly.GetManifestResourceStream("ClientCore.Resources.cncneticon.png");
+            using Stream tdiconStream = assembly.GetManifestResourceStream("ClientCore.Resources.tdicon.png");
+            using Stream raiconStream = assembly.GetManifestResourceStream("ClientCore.Resources.raicon.png");
+            using Stream unknowniconStream = assembly.GetManifestResourceStream("ClientCore.Resources.unknownicon.png");
+            using var dtaicon = Image.Load(dtaiconStream);
+            using var tiicon = Image.Load(tiiconStream);
+            using var tsicon = Image.Load(tsiconStream);
+            using var moicon = Image.Load(moiconStream);
+            using var yricon = Image.Load(yriconStream);
+            using var rricon = Image.Load(rriconStream);
+            using var cncricon = Image.Load(cncriconStream);
+            using var cncneticon = Image.Load(cncneticonStream);
+            using var tdicon = Image.Load(tdiconStream);
+            using var raicon = Image.Load(raiconStream);
+            using var unknownicon = Image.Load(unknowniconStream);
```

Comment:
Err kinda not fond of that code wall but I guess since the list is hardcoded there's no other way. Fix var names not adhering to camelCase though please.

Reply:
I would prefer defining a dictionary and looping on it, something like `"dtaicon":"ClientCore.Resources.dtaicon.png"`. This is called data-driven programming.